assume/strategies/learning_strategies.py

```diff
@@ -1006,19 +1035,30 @@ def create_observation(
         # =============================================================================
         # 1.1 Get the Observations, which are the basis of the action decision
         # =============================================================================
-        # scaling factors for the observations
-        scaling_factor_res_load = self.max_demand
-        scaling_factor_price = self.max_bid_price
+        # defines bounds of observation space
+        # stays here as it is unit specific, and different forecasts might apply for different units
+        # different handling would require an extra unit loop at learning role intiliazation and unit specific max/min values
+        # further forecasts might change during the simulation if advanced forecasting is used
+        self.max_market_price = max(unit.forecaster[f"price_{market_id}"])
+        self.min_market_price = min(unit.forecaster[f"price_{market_id}"])
+        self.max_residual = max(unit.forecaster[f"residualy_load_{market_id}"])
+        self.min_residual = min(unit.forecaster[f"residual_load_{market_id}"])
+
+        upper_scaling_factor_res_load = self.max_residual
+        lower_scaling_factor_res_load = self.min_residual
+        upper_scaling_factor_price = self.max_market_price
+        lower_scaling_factor_price = self.min_market_price
 
         # checks if we are at end of simulation horizon, since we need to change the forecast then
         # for residual load and price forecast and scale them
         if (
             end_excl + forecast_len
             > unit.forecaster[f"residual_load_{market_id}"].index[-1]
         ):
-            scaled_res_load_forecast = (
-                unit.forecaster[f"residual_load_{market_id}"].loc[start:]
-                / scaling_factor_res_load
+            scaled_res_load_forecast = min_max_scale(
+                unit.forecaster[f"residual_load_{market_id}"].loc[start:],
+                lower_scaling_factor_res_load,
+                upper_scaling_factor_res_load,
             )
             scaled_res_load_forecast = np.concatenate(
```

Comment:
Why no scaling in this part?

Reply:
I confusingly marked these lines, but I meant the ones below

Reply:
oh, this is a mistake, this should be fixed